app_architecture/ruby/ruby_on_rails.adoc

```diff
@@ -0,0 +1,332 @@
+
+----
+$ oc new-app path/to/source/code --name=rails-app
+----
+
+The command will check if there is a git repository configured, and will use the remote from that repository to configure the source from where OpenShift will download your source code for the build. Application is going to be named `rails-app`.
+
+The link:../../dev_guide/new_app.html[oc new-app] command takes either link to a remote `Git` repository or a local directory with your application from where the `oc` client will obtain the remote URL using `Git`.
+
+Once you have your application defined, you simply trigger the link:../../dev_guide/builds.html[build] process:
+
+----
+$ oc start-build rails-app
+----
+
+OpenShift is going to fetch the source code, setup the Builder image, build your application image and also will deploy the newly created image.
+
+Now, if you look at the running containers in OpenShift
+
+----
+$ oc get pods
+----
+
+You should see a line starting with myapp-(#number)-(some hash) and that is your application running in OpenShift.
+
+=== Creating database service
+
+Our Rails application expects a running database service, for which we will use link:http://www.postgresql.org/[PostgeSQL] database link:../../using_images/db_images/postgresql.html[image].
+
+Once again you will use the link:../../dev_guide/new_app.html[oc new-app] command and pass in necessary link:../../using_images/db_images/postgresql.html#environment-variables[environment variables] to be used inside the database container. These link:../../db_images/postgresql.html#environment-variables[environment variables] are required to set the username, password, and name of the database. You can change the values of these link:../../db_images/postgresql.html#environment-variables[environment variables] to anything you would like. The variables we are going to be setting are as follows:
+
+* POSTGRESQL_USER
+* POSTGRESQL_PASSWORD
+* POSTGRESQL_DATABASE
+
+By setting these variables when creating the PostgreSQL database, the image will ensure that:
+
+* A database exists with the specified name
+* A user exists with the specified name
+* The user can access the specified database with the specified password
+
+To do that type following command with your desired values:
+
+----
+$ oc new-app postgresql -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password -e POSTGRESQL_DATABASE=db_name
+----
+
+In case you would like to also set password for the database administrator, append the previous command with:
+----
+-e POSTGRESQL_ADMIN_PASSWORD=admin_pw
+----
+
+To watch the progress of this command type:
+
+----
+$ oc get pods --watch
+----
+
+=== Wiring the frontend service to communicate with database
+
+When we initially created our Ruby on Rails application, we provided no environment variables with informations about database. The application is looking for a database, but can't find one, and it fails.
```

Comment:
i think it would be better to reverse the order...create the DB first, then the app, so you can eliminate this step.